pkg/security/probe/resolvers/mount_resolver.go

```diff
@@ -418,13 +400,36 @@ func (mr *MountResolver) ResolveMountPath(mountID, pid uint32, containerID strin
 	return mr.resolveMountPath(mountID, pid, containerID)
 }
 
+func (mr *MountResolver) isSyncCacheAllowed(mountID uint32) bool {
+	now := time.Now()
+	if ts, ok := mr.fallbackLimiter.Get(mountID); ok {
+		if now.After(ts) {
+			mr.fallbackLimiter.Remove(mountID)
+		} else {
+			return false
+		}
+	}
+	return true
+}
+
+func (mr *MountResolver) syncCacheMiss(mountID uint32) {
+	mr.procMissStats.Inc()
+
+	// add to fallback limiter to avoid storm of file access
+	mr.fallbackLimiter.Add(mountID, time.Now().Add(fallbackLimiterPeriod))
+}
+
 func (mr *MountResolver) resolveMountPath(mountID, pid uint32, containerID string) (string, error) {
 	if _, err := mr.IsMountIDValid(mountID); err != nil {
 		return "", err
 	}
 
 	// force pid1 resolution here to keep the LRU doing his job and not evicting important entries
 	pid1, _ := mr.cgroupsResolver.GetPID1(containerID)
+	if pid1 == 0 {
+		// use the pid1 of the host
+		pid1 = 1
+	}
```

Comment:
Shouldn't we move the pid1 resolution after the `if !mr.opts.UseProcFS` return block in both `resolveMountPath` and `resolveMount`?

Reply:
thanks, I moved it